resources/assets/js/components/HRPortal/JobIndexHrPage.tsx

```diff
@@ -256,15 +253,15 @@ const JobIndexHrDataFetcher: React.FC<JobIndexHrDataFetcherProps> = ({
 
   // Request and select all managers belonging to the dept jobs
   const managers = useSelector(getManagers);
-  const managersUpdating = useSelector(getManagerIsUpdatingById);
-  deptJobs.forEach((job: Job): void => {
-    if (
-      find(managers, job.manager_id) === null &&
-      managersUpdating[job.manager_id] !== true
-    ) {
-      dispatch(fetchManager(job.manager_id));
-    }
-  });
+  useEffect(() => {
+    const uniqueManagers: number[] = [];
+    deptJobs.forEach((job: Job): void => {
+      if (!uniqueManagers.includes(job.manager_id)) {
```

Comment:
I think this check can still include `find(managers, job.manager_id) === null`. No need to dispatch a fetch if we already have the data.

Reply:
Well... We tried to change this but kept running into issues where modifying a hook dependency (ie `managers`) caused the whole loop to run again. We'll leave as is for now.